### DISCUSSION TITLE
How does seek() works on S3?

### DISCUSSION THREAD

**Discussion by shay1bz** [RaRe-Technologies/smart_open#708] [Q&A] [answered]
Hi,

I'm trying to find a solution for random access inside large Avro files on S3 with Python. Does smart_open download the entire file to enable seek(), or does it really "seek" (via http headers I guess) to the desired location? In my case the files are about 200-500MB, and the casual reader would read a very small portion of it by a given start/end offsets.

**Comment by mpenkov** [comment, marked as answer]
It uses HTTP headers.

**Reply by shay1bz** [reply]
Thank you
